ap_src/ap_app/static/js/button_spinner.js

```diff
@@ -0,0 +1,42 @@
+/*Logic for spinner for create team button -KJ*/
+document.addEventListener('DOMContentLoaded', function() {
```

Comment:
You should be able to simplify this whole block of code with something like (note, I've not tested this so might need some tweaking):

```
document.addEventListener('DOMContentLoaded', function() {
    var nameInput = document.getElementById('nameInput');
    var descriptionInput = document.getElementById('id_description');
    var submitButton = document.getElementById('submit');
    var labelButton = document.getElementById('label_button');

    function updateButtonState() {
        const nameValue = nameInput.value.trim();
        const descriptionValue = descriptionInput.value.trim();
        const shouldEnable = nameValue !== "" && descriptionValue !== "";

        if (shouldEnable) {
            submitButton.removeAttribute("disabled");
            labelButton.removeAttribute("disabled");
        } else {
            submitButton.setAttribute("disabled", true);
            labelButton.setAttribute("disabled", true);
        }
    }

    nameInput.addEventListener("keyup", updateButtonState);
    descriptionInput.addEventListener("keyup", updateButtonState);

    labelButton.addEventListener('click', () => {
        if (!labelButton.hasAttribute("disabled")) {
            labelButton.classList.add('is-loading');
            labelButton.setAttribute('disabled', true);
        }
    });
});
```